lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -47,15 +47,15 @@ def deploy_failed?
     end
 
     def failure_message
-      @latest_rs.failure_message
+      @latest_rs&.failure_message
```

Comment:
It is unlikely but possible that no replicaset will be found. That's why we have `return false unless @latest_rs` in other spots. I'm not sure what made me notice these guys now, but we shouldn't assume we have an RS here either.